drivers/net/ethernet/intel/idpf/eth/idpf_netdev.h

```diff
@@ -0,0 +1,82 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/* Copyright (C) 2024 Intel Corporation */
+
+#ifndef _IDPF_NETDEV_H_
+#define _IDPF_NETDEV_H_
+
+/**
+ * struct idpf_netdev_priv - Struct to store vport back pointer
+ * @adapter: Adapter back pointer
+ * @vport: Vport back pointer
+ * @vport_id: Vport identifier
+ * @vport_idx: Relative vport index
+ * @state: See enum idpf_vport_state
+ * @netstats: Packet and byte stats
+ * @stats_lock: Lock to protect stats update
+ */
+struct idpf_netdev_priv {
```

Comment:
You can keep one of the two here : vport or adapter. Both should not be needed.
From adapter you can reach vport   adapter->vport
From vport you can reach adapter    vport->adapter

Reply:
This is the old structure, I did not add this in this PR. I think this should be done in separate PR if needed.